includes/class-wc-payments-account.php

```diff
@@ -926,169 +1101,156 @@ public function maybe_handle_onboarding() {
-						);
-					}
-
-					// We should not reach this point as we either redirect to the Stripe KYC
-					// or to the Connect page with an error message.
-					return;
-				}
+			// First requirement: handle the WPCOM/Jetpack connection.
+			// If there is a working one, we can proceed with the Stripe account handling.
+			try {
+				$this->maybe_init_jetpack_connection(
+					// Carry over all the important GET params, so we have them after the Jetpack connection setup.
+					add_query_arg(
+						[
+							'promo'                  => ! empty( $incentive_id ) ? $incentive_id : false,
+							'progressive'            => $progressive ? 'true' : false,
+							'create_builder_account' => $create_builder_account ? 'true' : false,
+							'test_mode'              => $should_onboard_in_test_mode ? 'true' : false,
+							'from'                   => WC_Payments_Onboarding_Service::FROM_WPCOM_CONNECTION,
+							'source'                 => $onboarding_source,
+
+						],
+						self::get_connect_url( $wcpay_connect_param ) // Instruct Jetpack to return here (connect link).
+					)
+				);
+			} catch ( API_Exception $e ) {
+				Logger::error( 'Init Jetpack connection failed. ' . $e->getMessage() );
+				$this->redirect_service->redirect_to_connect_page(
+				/* translators: %s: error message. */
+					sprintf( __( 'There was a problem connecting your store to WordPress.com: "%s"', 'woocommerce-payments' ), $e->getMessage() ),
+					WC_Payments_Onboarding_Service::FROM_WPCOM_CONNECTION,
+					[ 'source' => $onboarding_source ]
+				);
+				return;
 			}
 
 			// Handle the flow for [a builder] moving from test to live.
 			if ( ! empty( $_GET['wcpay-disable-onboarding-test-mode'] ) && 'true' === $_GET['wcpay-disable-onboarding-test-mode'] ) {
-				$test_mode = WC_Payments_Onboarding_Service::is_test_mode_enabled();
-
 				// Delete the account if the test mode is enabled,
 				// otherwise it'll cause issues when trying to onboard again.
-				if ( $test_mode ) {
-					// Delete the account.
-					$this->payments_api_client->delete_account( $test_mode );
+				if ( WC_Payments_Onboarding_Service::is_test_mode_enabled() ) {
+					// Delete the currently Stripe connected account.
+					$this->payments_api_client->delete_account( true );
 					// Make sure we clear the cached account data as it may take a while
 					// for the account update webhook to come from our platform.
 					$this->clear_cache();
 				}
 
-				// Set the test mode to false now that we are handling a real onboarding.
+				// Disable test mode onboarding now that we are handling a live onboarding.
 				WC_Payments_Onboarding_Service::set_test_mode( false );
 
-				$this->redirect_to_onboarding_page_or_start_server_connection( $onboarding_source );
-				return;
+				// Only redirect to the onboarding wizard if the new onboarding flow is enabled.
+				// Otherwise, proceed to the Stripe init.
+				if ( WC_Payments_Utils::should_use_new_onboarding_flow() ) {
```

Comment:
it's a bit confusing IMO cause we use new onboarding flow by default now

Reply:
I looked into the history of this - it was introduced in #7773, and I think our thinking was to leave a way of using the old onboarding **just in case** there was a bug with the new onboarding that meant we needed to provide a workaround.

At this point, it's been 8 months and we haven't needed to use it, so my opinion is we should clean it up and remove it (in a new issue, since I don't want to add more to this PR 🙂 )

Reply:
Good call, Daniel. I am all for removing stuff. Could you wrangle and issue and add it to our backlog?

Reply:
No need for another issue. I've removed it in this PR :)

Reply:
Oh, awesome. Thank you for taking care of it!